pkg/config/expand.go

```diff
@@ -61,10 +61,44 @@ func (dc *DeploymentConfig) expand() error {
 		return err
 	}
 
+	if err := validateModulesInputs(dc.Config); err != nil {
+		return err
+	}
+
 	dc.Config.populateOutputs()
 	return nil
 }
 
+func validateModulesInputs(bp Blueprint) error {
+	errs := Errors{}
+	for ig, g := range bp.DeploymentGroups {
+		for im, m := range g.Modules {
+			p := Root.Groups.At(ig).Modules.At(im)
+			errs.Add(validateModuleInputs(p, m, bp))
+		}
+	}
+	return errs.OrNil()
+}
+
+func validateModuleInputs(mp modulePath, m Module, bp Blueprint) error {
```

Comment:
nit: The names seem kind of confusing at first glance (it becomes more obvious after looking at the code though).  Maybe these names might be a little bit easier to understand?

```suggestion
func validateInputsAllModules(bp Blueprint) error {
	errs := Errors{}
	for ig, g := range bp.DeploymentGroups {
		for im, m := range g.Modules {
			p := Root.Groups.At(ig).Modules.At(im)
			errs.Add(validateModuleInputs(p, m, bp))
		}
	}
	return errs.OrNil()
}

func validateModuleInputs(mp modulePath, m Module, bp Blueprint) error {
```